src/Http/Routing/src/Matching/HttpMethodDictionaryPolicyJumpTable.cs

```diff
@@ -21,24 +114,48 @@ internal sealed class HttpMethodDictionaryPolicyJumpTable : PolicyJumpTable
         Dictionary<string, int>? corsPreflightDestinations)
     {
         _exitDestination = exitDestination;
+        _knownHttpMethodDestinations = new HttpMethodsJumpTable(exitDestination);
+        _corsPreflightHttpMethodDestinations = new HttpMethodsJumpTable(corsPreflightExitDestination);
+        if (destinations != null)
+        {
+            foreach (var item in destinations)
+            {
+                _knownHttpMethodDestinations.Set(item.Key, item.Value);
+            }
+        }
```

Comment:
Most of the time, the destinations/corsPreflightDestinations dictionaries will only contain known destinations. Keeping the dictionary around becomes unnecessary. It would reduce memory usage to never save the dictionary on the jump table.

I wonder if `HttpMethodsJumpTable` could be refactored into something that the builder uses directly [here](https://github.com/dotnet/aspnetcore/blob/7f18f8fea5c8e2efc26050f0815f8c911bb26ff1/src/Http/Routing/src/Matching/HttpMethodMatcherPolicy.cs#L311-L386) and is then passed to the jump table. There is a set method that either sets the value directly if it is a known header value, or sets it into a lazily created dictionary if it is unknown. That would avoid creating a dictionary at all during startup most of the time.

Reply:
That is a good point. Let me have a look and update the PR.